lib/lhm/entangler.rb

```diff
@@ -78,14 +82,16 @@ def validate
     end
 
     def before
+      kill_long_running_queries_on_origin_table if special_origin?
       entangle.each do |stmt|
-        @connection.execute(tagged(stmt))
+        execute_with_timeout(stmt, MAX_RUNNING_SECONDS)
```

Comment:
There is a race condition here, where a long-running query _could_ start in between killing the ones that were already there, and starting the trigger creation / deletion.

My first thought for how to fix that is: start the trigger creation in a thread, then kill long-running queries on origin table after a few seconds.